scrips/eventsData.js

```diff
@@ -0,0 +1,132 @@
+import { getLongWeekDayName, getLongMonthName } from "./utils.js";
+import { cellHeight, currentEventTileId, defaultEventLength, emptyEventTitle } from "./calendarVars.js";
+import { placeNewEventTile, createEventTile } from "./eventTile.js";
+import { getModalInputs, getGridDays } from "./selectors.js";
+import { closeModal } from "./modal.js";
+import { minutesToHour, hoursToMinutes, formatMinutes, formatHours } from "./timeCalculations.js";
+import {
+    currentEventDataKey,
+    getDataFromLocalStorage,
+    savedEventsKey,
+    storeDataInLocalStorage,
+} from "./handleLocalStorage.js";
+
+const adjustEventTopPosition = (event) => {
+    const distanceFromTop = event.target.getBoundingClientRect().top;
+    const clickPosition = event.clientY - distanceFromTop;
+    const increment = cellHeight / 2;
+
+    return Math.floor(clickPosition / increment) * increment;
+};
+
+const getEventLength = (eventData) => {
+    const startHour = parseInt(eventData.startTime.hour, 10);
+    const startMin = minutesToHour(eventData.startTime.minutes);
+    const endHour = parseInt(eventData.endTime.hour, 10);
+    const endMin = minutesToHour(eventData.endTime.minutes);
+
+    return endHour + endMin - startHour - startMin;
+};
+
+const setTime = (hour, minutes) => {
+    return {
+        hour: `${formatHours(hour)}`,
+        minutes: `${formatMinutes(minutes)}`,
+    };
+};
+
+const setDefaultEndTime = (startHour, minutes, eventLength = defaultEventLength) => {
+    const endTime = parseInt(startHour, 10) + minutesToHour(parseInt(minutes, 10)) + eventLength;
+    const endHour = Math.floor(endTime);
+    const endMinutes = hoursToMinutes(endTime - endHour);
+
+    return {
+        hour: `${formatHours(endHour)}`,
+        minutes: `${formatMinutes(endMinutes)}`,
+    };
+};
+
+const constructNewEvent = (event) => {
+    const clickedColumn = event.target;
+    const clickPosition = adjustEventTopPosition(event);
+    const startHour = Math.floor(clickPosition / cellHeight);
+    const weekdayShort = clickedColumn.getAttribute("data-weekday");
+    const monthNameShort = clickedColumn.getAttribute("data-month");
+
+    let minutes = 0;
+    if (clickPosition % cellHeight === cellHeight / 2) minutes = 30;
+
+    return {
+        title: emptyEventTitle,
+        startTime: setTime(startHour, minutes),
+        endTime: setDefaultEndTime(startHour, minutes),
+        weekday: weekdayShort,
+        weekdayLong: getLongWeekDayName(weekdayShort),
+        day: clickedColumn.getAttribute("data-day"),
+        month: monthNameShort,
+        monthLong: getLongMonthName(monthNameShort),
+        year: clickedColumn.getAttribute("data-year"),
+    };
+};
+
+const recordNewEvent = (eventData) => {
```

Comment:
I guess you are saving event here

```suggestion
const saveNewEvent = (eventData) => {
```